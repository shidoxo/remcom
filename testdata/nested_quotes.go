package main

import "fmt"

func main() {
	// Complex nesting scenarios
	nested1 := "Outer \"inner // not a comment\" outer" // Real comment
	nested2 := `Raw "with quotes" and // fake comment`  // Another real comment

	// Multiple quote types
	mixed1 := "Double quote with 'single' inside"     // Mixed quotes comment
	mixed2 := `Raw with "double" and 'single' quotes` // Raw mixed comment

	// Tricky escape combinations
	tricky1 := "End with quote: \""      // Comment after quote
	tricky2 := "Escaped backslash: \\\\" // Comment after escapes
	tricky3 := "Escaped quote: \\\""     // Comment after escaped quote

	// Very complex nesting
	complex := "Start \"middle 'inner // fake' middle\" end" // Real comment

	// JSON-like strings
	jsonLike := `{"url": "https://api.com/endpoint", "comment": "// not real"}` // JSON comment

	// Code-like strings
	codeLike := "func main() { // this is fake code }" // Code comment

	// Edge case: quote at very end
	endQuote := "String ending with quote \"" // End quote comment

	fmt.Println(nested1, nested2)          // Print nested
	fmt.Println(mixed1, mixed2)            // Print mixed
	fmt.Println(tricky1, tricky2, tricky3) // Print tricky
	fmt.Println(complex)                   // Print complex
	fmt.Println(jsonLike)                  // Print JSON-like
	fmt.Println(codeLike)                  // Print code-like
	fmt.Println(endQuote)                  // Print end quote
}
