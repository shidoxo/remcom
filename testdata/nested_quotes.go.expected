package main

import "fmt"

func main() {
	nested1 := "Outer \"inner // not a comment\" outer"
	nested2 := `Raw "with quotes" and // fake comment`

	mixed1 := "Double quote with 'single' inside"
	mixed2 := `Raw with "double" and 'single' quotes`

	tricky1 := "End with quote: \""
	tricky2 := "Escaped backslash: \\\\"
	tricky3 := "Escaped quote: \\\""

	complex := "Start \"middle 'inner // fake' middle\" end"

	jsonLike := `{"url": "https://api.com/endpoint", "comment": "// not real"}`

	codeLike := "func main() { // this is fake code }"

	endQuote := "String ending with quote \""

	fmt.Println(nested1, nested2)
	fmt.Println(mixed1, mixed2)
	fmt.Println(tricky1, tricky2, tricky3)
	fmt.Println(complex)
	fmt.Println(jsonLike)
	fmt.Println(codeLike)
	fmt.Println(endQuote)
}
